### ISSUE TITLE
Mosaic Augment in Image Classification? 

### ISSUE THREAD

**Issue by innat** [AlexeyAB/darknet#7088]
Hello @AlexeyAB 

I'm trying to use **Mosaic Augmentation** for image classification. But I'm facing an issue to create the augmented class label. As the methodology is inspired by your work, I need some advice of yours on this. I've asked a question in [SO](https://stackoverflow.com/questions/65181294/how-to-create-class-label-for-mosaic-augmentation-in-image-classification), if you have time, would you please check? Thanks.

**Comment by stephanecharette** [comment]
Are you trying to do mosaic by hand?  Instead, set `mosaic=1` in the `[net]` section of the .cfg file.  You don't have to do it by hand.

**Comment by innat** [comment]
Sorry, I think you misunderstood. I'm not using an object detection task, it's a classification. So, I shouldn't have to bother with the `net` section in `.ctf`.  Please follow the given link above. I'm asking here because this augmentation idea is from `Yolo4` paper-work.

**Comment by innat** [comment]
@AlexeyAB Any suggestion? Otherwise, I like to close it.

**Comment by AlexeyAB** [comment]
You can use `[net] mosaic=1` for training Classifier too: https://github.com/AlexeyAB/darknet/blob/4ee3be7e68fb9c7eda5cc390e47e59f01e40dded/src/data.c#L1913-L1942

Wiki: https://github.com/AlexeyAB/darknet/wiki/CFG-Parameters-in-the-%5Bnet%5D-section

**Comment by innat** [comment]
Thanks for the pointer. However, uncommented code is typically hard to follow. Is there any detailed explanation of how the label is generated in mosaic?

**Comment by kairavpatel** [comment]
@AlexeyAB , Mosaic data augmentation technique first randomly merge 4 images and then crop single random patch from the merged data?? right, If not please tell me how basically work.

**Comment by stephanecharette** [comment]
For mosaic, see the paper here:  https://arxiv.org/pdf/2004.10934.pdf
Example images can be seen here:  https://www.ccoderun.ca/darkmark/DataAugmentationMisc.html#Mosaic

**Comment by kairavpatel** [comment]
> For mosaic, see the paper here: https://arxiv.org/pdf/2004.10934.pdf
> Example images can be seen here: https://www.ccoderun.ca/darkmark/DataAugmentationMisc.html#Mosaic

Thank you for replying. But, I want know, 4 images are randomly mixed and generate a single image. Or, it removes the 4 parts of the single image and adds four different images in that part like CUTMix...??.

**Comment by AlexeyAB** [comment]
> 4 images are randomly mixed and generate a single image.

This is used for Detector.

----

> it removes the 4 parts of the single image and adds four different images in that part like CUTMix

This is used for Classifier.

----

But in general, they are about the same.

**Comment by kairavpatel** [comment]
> > 4 images are randomly mixed and generate a single image.
> 
> This is used for Detector.
> 
> > it removes the 4 parts of the single image and adds four different images in that part like CUTMix
> 
> This is used for Classifier.
> 
> But in general, they are about the same.

Thank you for the reply..
